src/lib/core/say.c

```diff
@@ -710,6 +723,8 @@ log_create(struct log *log, const char *init_str, int nonblock)
 	} else {
 		log->type = SAY_LOGGER_STDERR;
 		log->fd = STDERR_FILENO;
+		log->isatty = isatty(log->fd);
```

Comment:
Why did you prefer to add this chunk individually to each function in the switch above and then added it here. I suppose, it is better to move it right after this if statement.

Reply:
corrected